dbt/adapters/snowflake/impl.py

```diff
@@ -99,15 +102,20 @@ def _use_warehouse(self, warehouse: str):
     def pre_model_hook(self, config: Mapping[str, Any]) -> Optional[str]:
         default_warehouse = self.config.credentials.warehouse
         warehouse = config.get("snowflake_warehouse", default_warehouse)
+        logger.info("Running pre_model_hook:")
+        logger.info(f"Default warehouse: {default_warehouse}, Selected warehouse: {warehouse}")
         if warehouse == default_warehouse or warehouse is None:
             return None
         previous = self._get_warehouse()
         self._use_warehouse(warehouse)
+        logger.info(f"Changed wareho use from {previous} to {warehouse}")
```

Comment:
Do we want to put these logs behind condition of warehouses being different?

Reply:
@McKnight-42 I don't think we should add any new logging here. The actual queries being run (`select current_warehouse() as warehouse` + `use warehouse different_warehouse`) are already appearing in the logs.